### PR TITLE
Add Programvareverkstedet (pvv.ntnu.no, pvv.org)

### PR DESCRIPTION
**Pull request by jornane**
Hey everyone!

I'd like to add this rule to HTTPS Everywhere

### CODE REVIEW

**Comment by Bisaloo** [comment]
Hi @jornane, thank you for your PR!

Please have a look at our [contributing guide](https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md#wildcards-in-targets), and in the particular the section I linked about wildcards. To sum up, our new guidelines recommend enumerating individual subdomain whenever possible.

**Comment by jornane** [comment]
Thanks @Bisaloo, does this look better?

File: src/chrome/content/rules/pvv.ntnu.no.xml [hunk 1]
```diff
@@ -1,20 +1,17 @@
 <ruleset name="Programvareverkstedet">
-	<target host="*.pvv.ntnu.no" />
-	<target host="*.pvv.org" />
+	<target host="http://www.pvv.org" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/pvv.ntnu.no.xml]: `target`s should only contain the hostname, not the protocol

```suggestion
	<target host="www.pvv.org" />
```

File: src/chrome/content/rules/pvv.ntnu.no.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	<target host="http://mail.pvv.org" />
+	<target host="http://mail.pvv.ntnu.no" />
+	<target host="http://webmail.pvv.org" />
+	<target host="http://webmail.pvv.ntnu.no" />
+	<target host="http://idp.pvv.org" />
+	<target host="http://idp.pvv.ntnu.no" />
+	<target host="http://list.pvv.org" />
+	<target host="http://list.pvv.ntnu.no" />
+
+	<rule from="^http://(.+\.)?pvv\.org" to="https://$1.pvv.ntnu.no" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/pvv.ntnu.no.xml]: Please try use use a stricter regex pattern, such as:

```suggestion
	<rule from="^http://(\w+\.)?pvv\.org" to="https://$1pvv.ntnu.no" />
```

You also have an extra `.` in your `to`.

File: src/chrome/content/rules/pvv.ntnu.no.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	<target host="mail.pvv.org" />
+	<target host="mail.pvv.ntnu.no" />
+	<target host="webmail.pvv.org" />
+	<target host="webmail.pvv.ntnu.no" />
+	<target host="idp.pvv.org" />
+	<target host="idp.pvv.ntnu.no" />
+	<target host="list.pvv.org" />
+	<target host="list.pvv.ntnu.no" />
+
+	<rule from="^http://(\w+\.)?pvv\.org" to="https://$1pvv.ntnu.no" />
```
Comment (Bisaloo) [inline src/chrome/content/rules/pvv.ntnu.no.xml]: You're missing trailing slashes here
```suggestion
	<rule from="^http://(\w+\.)?pvv\.org/" to="https://$1pvv.ntnu.no/" />
```

**Comment by Bisaloo** [comment]
Please also see my comment above about file name and target ordering.

Anyways, looks like we're gonna have to put this on hold since many certificates expired since you submitted this PR. Let's see if they renew them.

**Comment by pipboy96** [comment]
@jornane Are you an owner of this website?

**Comment by jornane** [comment]
> @jornane Are you an owner of this website?

It's a student project. I'm helping. I see some certificate didn't make it in when I renewed last time, I'll check that out. Is there a way to rebuild, or should I push an empty commit? :/

**Comment by pipboy96** [comment]
@jornane try doing `git commit --amend` without actually changing anything, and force-pushing afterwards, or just do `git commit --allow-empty -m 'Rerun Travis'`, at your choice.

**Comment by jornane** [comment]
% git commit --amend
	On branch pvv
	Your branch is up to date with 'pr/pvv'.

	No changes
	You asked to amend the most recent commit, but doing so would make
	it empty. You can repeat your command with --allow-empty, or you can
	remove the commit entirely with "git reset HEAD^".

Empty commit it is, then.

**Comment by pipboy96** [comment]
@jornane I think this can be prevented by adding `--no-edit`. I guess it's better to not force-push without reason, so adding an empty commit is a better choice.

**Comment by Bisaloo** [comment]
Great! One last thing I forgot to mention: can you please document all non-working hosts (both the ones you fix with the ruleset and the others, such as `spikkjeposche.pvv.org`): https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md#non-working-hosts

(FYI, you can also restart travis by closing/re-opening your PR and by asking someone with write access.)

**Comment by jornane** [comment]
> can you please document all non-working hosts

What do you mean with non-working hosts? I've added some hosts, but I'm not sure what you mean with "non-working hosts". The document mentions "Invalid certificate", "Redirect to HTTP", "Refused" and "Time out". I don't think we have any of those.

**Comment by Bisaloo** [comment]
For example, all `*.pvv.org` targets have an invalid certificate. Hence your special rule on L29.

**Comment by jornane** [comment]
Do you recommend a listing like in the documentation, or simply a statement that certificates contain `pvv.ntnu.no` names but not `pvv.org`? I've added a statement, let me know if I should add a list (it will be a copy of all targets with names under `pvv.org`).

**Comment by Bisaloo** [comment]
@jornane, something strange is going on with the certs. It was failing locally so I restarted travis and it fails again. Could you look into it please?

**Comment by jornane** [comment]
Ok, attempting close/reopen.

**Comment by Bisaloo** [comment]
Thank you!

**Pull request merged**